provisioner/src/lib.rs

```diff
@@ -457,6 +555,301 @@ impl Provisioner for MyProvisioner {
 
         Ok(Response::new(reply))
     }
+
+    #[tracing::instrument(skip(self))]
+    async fn provision_dynamo_db(
+        &self,
+        request: Request<DynamoDbRequest>,
+    ) -> Result<Response<DynamoDbResponse>, Status> {
+        verify_claim(&request)?;
+
+        let request = request.into_inner();
+
+        let reply = self.request_dynamodb(&request.project_name).await?;
+
+        Ok(Response::new(reply))
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn delete_dynamo_db(
+        &self,
+        request: Request<DynamoDbRequest>,
+    ) -> Result<Response<DynamoDbDeletionResponse>, Status> {
+        verify_claim(&request)?;
+
+        let request = request.into_inner();
+
+        let reply = self.delete_dynamodb(&request.project_name).await?;
+
+        Ok(Response::new(reply))
+    }
+}
+
+struct DynamoDBHandler {
```

Comment:
I am thinking if it will be a good idea to separate Identity, policy creation and management part from this Handler, so that can be independently re-used for other future AWS resouces.
For one such instance - while working on s3 resource PR, [s3-resource](https://github.com/shuttle-hq/shuttle/pull/995/files#diff-9aa440c73eedfd63ea9f7221d4be2d948eb7af1bffaadd9eec69a379ab1c9cadR630) I am doing similar stuff of creating identity, creating policy and attaching.